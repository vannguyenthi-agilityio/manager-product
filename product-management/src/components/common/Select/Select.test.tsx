//Libraries
import { fireEvent, render, screen } from '@testing-library/react';

//Components
import { Select } from '.';

const mockOptions = [
  { label: 'Option 1', value: '1' },
  { label: 'Option 2', value: '2' },
  { label: 'Option 3', value: '3' }
];

describe('Select render', () => {
  test('match with snapshot', () => {
    render(
      <Select
        options={mockOptions}
        value={mockOptions[0].value}
        label='Select'
        onChange={() => {}}
      />
    );

    const select = screen.getByTestId('select');

    expect(select).toMatchSnapshot();
  });

  test('should render without label', () => {
    render(
      <Select
        options={mockOptions}
        value={mockOptions[0].value}
        onChange={() => {}}
      />
    );

    const select = screen.queryByText('Select');

    expect(select).toBeFalsy();
  });

  test('should simulate onChange event', () => {
    const mockOnChange = jest.fn();

    render(
      <Select
        options={mockOptions}
        value={mockOptions[0].value}
        label='Select'
        onChange={mockOnChange}
      />
    );

    const select = screen.getByTestId('select');
    fireEvent.change(select);

    expect(mockOnChange).toHaveBeenCalled();
  });
});
