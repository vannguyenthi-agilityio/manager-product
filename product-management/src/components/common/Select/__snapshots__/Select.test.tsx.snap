// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select render match with snapshot 1`] = `
<select
  class="chakra-select css-1ik61og"
  data-testid="select"
>
  <option
    value="1"
  >
    Option 1
  </option>
  <option
    value="2"
  >
    Option 2
  </option>
  <option
    value="3"
  >
    Option 3
  </option>
</select>
`;
